third_party/docfx-doclet-143274/src/main/java/com/microsoft/util/Utils.java

```diff
@@ -250,4 +252,19 @@ public List<Element> getImplementedMethods(
   public List<TypeElement> getImplementedInterfaces(TypeElement element) {
     return element.getInterfaces().stream().map(e -> asTypeElement(e)).collect(Collectors.toList());
   }
+
+  // Use to get the recommended package URL for Package Overview
+  public static String[] extractPackageBaseURIBeforeVersion(String input) {
+    // This regex captures the package path before the version (if it exists) and the version
+    Pattern pattern = Pattern.compile("(.*?)(v\\d+.*?)(?:\\.|$)");

```

Comment:
`Pattern.compile` is an expensive operation recommended to be done only once rather than every time it's used -- hence the way it's been split by `compile` and then `matcher`.

Simple edit: move this line to a static field of this class.

Reply:
Ah good to know - thank you! Refactored.